src/main/java/com/projecturanus/betterp2p/client/gui/widget/WidgetButton.kt

```diff
@@ -0,0 +1,69 @@
+package com.projecturanus.betterp2p.client.gui.widget
+
+import com.projecturanus.betterp2p.MODID
+import com.projecturanus.betterp2p.client.gui.GUI_TEX_HEIGHT
+import com.projecturanus.betterp2p.client.gui.GUI_WIDTH
+import com.projecturanus.betterp2p.client.gui.GuiAdvancedMemoryCard
+import com.projecturanus.betterp2p.client.gui.drawTexturedQuad
+import net.minecraft.client.Minecraft
+import net.minecraft.client.gui.GuiButton
+import net.minecraft.client.renderer.OpenGlHelper
+import net.minecraft.client.renderer.Tessellator
+import org.lwjgl.opengl.GL11
+
+/**
+ * Widget button for stuff.
+ * @param x
+ * @param y
+ * @param width
+ * @param height
+ * @param hoverText - unlocalized text to display when mouse hovering
+ */
+abstract class WidgetButton(val gui: GuiAdvancedMemoryCard, x: Int, y: Int, width: Int, height: Int,
+                            val hoverText: () -> String): GuiButton(0, x, y, width, height, "") {
+
+    private var texX = 0.0
+    private var texY = 0.0
+
+    abstract fun mousePressed(mouseX: Int, mouseY: Int)
+
+    fun draw(mc: Minecraft?, mouseX: Int, mouseY: Int, partialTicks: Float) {
+        val tessellator = Tessellator.instance
+        gui.bindTexture(gui.BACKGROUND)
+        GL11.glColor4f(1.0f, 1.0f, 1.0f, 1.0f)
+        field_146123_n = mouseX >= xPosition && mouseY >= yPosition && mouseX < xPosition + width && mouseY < yPosition + height
+        val k = getHoverState(field_146123_n)
+        GL11.glEnable(GL11.GL_BLEND)
+        OpenGlHelper.glBlendFunc(770, 771, 1, 0)
```

Comment:
770 = GL11.GL_SRC_ALPHA
771 = GL11.GL_ONE_MINUS_SRC_ALPHA
iirc but double check, the line below is not needed either

Reply:
Fixed.